framework/core/src/User/Command/RequestPasswordResetHandler.php

```diff
@@ -90,29 +64,19 @@ public function handle(RequestPasswordReset $command)
             ['email' => 'required|email']
         );
 
-        if ($validation->fails()) {
-            throw new ValidationException($validation);
-        }
-
         $user = $this->users->findByEmail($email);
 
-        if (! $user) {
-            throw new ModelNotFoundException;
+        // Prevents leaking user existence.
```

Comment:
How does this not leak user existence?

We still throw an error if the user is not found, it just happens to be a validation error instead. This too causes the issue of confusing genuine users who have entered a valid email and are then told the email entered isn't valid.
 
If a user _knows_ a valid email is entered, they will still know that the user does/doesn't exist based on if we still return an error or not.

Wouldn't it be better to not return an error and instead use the ever more popular approach of "if this user exists, we'll send them an email"?

Reply:
🤦🏼 you're totally right